src/one-source/buffer-with-time-or-count.js

```diff
@@ -0,0 +1,63 @@
+const {createStream, createProperty} = require('../patterns/one-source');
+
+const mixin = {
+
+  _init({wait, count, flushOnEnd}) {
+    this._wait = wait;
+    this._count = count;
+    this._flushOnEnd = flushOnEnd;
+    this._intervalId = null;
+    this._$onTick = () => this._flush();
+    this._buff = [];
+  },
+
+  _free() {
+    this._$onTick = null;
+    this._buff = null;
+  },
+
+  _flush() {
+    if (this._buff !== null && this._buff.length !== 0) {
```

Comment:
Sure - do you want to still avoid flushing an empty buffer on end, though? This isn't a common case with the other multi-source buffer methods because end typically hits right after some value. Here though, since we flush the buffer immediately any time count is reached, we'll emit an empty array on end whenever the source observable has `#events in last stretch % count = 0`. For instance, case 2 in the docs will become:

```
  source:  ---1---2---3---4---5---6---7---8  X
  result:  -------•-------•-------•-------•  X
              [1,2]   [3,4]   [5,6]   [7,8] []
```

This would also occur with the other buffer methods if end arrived substantially later than the final value, but it's a much more common corner case here since we handle the flush condition synchronously before processing end.